<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>mwin</class>
 <widget class="QMainWindow" name="mwin">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1202</width>
    <height>993</height>
   </rect>
  </property>
  <property name="mouseTracking">
   <bool>false</bool>
  </property>
  <property name="windowTitle">
   <string>mwin</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QSpinBox" name="repl_cu_select">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>770</y>
      <width>47</width>
      <height>23</height>
     </rect>
    </property>
    <property name="minimum">
     <number>0</number>
    </property>
    <property name="maximum">
     <number>99</number>
    </property>
    <property name="singleStep">
     <number>1</number>
    </property>
   </widget>
   <widget class="QLabel" name="repl_cu_select_label">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>770</y>
      <width>21</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>CU:</string>
    </property>
   </widget>
   <widget class="QFrame" name="frame">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>800</y>
      <width>371</width>
      <height>71</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QPushButton" name="criar_cu">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>111</width>
       <height>22</height>
      </rect>
     </property>
     <property name="text">
      <string>add control unit</string>
     </property>
    </widget>
    <widget class="QPushButton" name="criar_bus">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>40</y>
       <width>111</width>
       <height>22</height>
      </rect>
     </property>
     <property name="text">
      <string>add bus</string>
     </property>
    </widget>
    <widget class="QLabel" name="repl_cu_select_label_2">
     <property name="geometry">
      <rect>
       <x>240</x>
       <y>10</y>
       <width>31</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>BUS:</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="repl_bus_select">
     <property name="geometry">
      <rect>
       <x>270</x>
       <y>10</y>
       <width>47</width>
       <height>23</height>
      </rect>
     </property>
     <property name="minimum">
      <number>0</number>
     </property>
     <property name="maximum">
      <number>99</number>
     </property>
     <property name="singleStep">
      <number>1</number>
     </property>
    </widget>
    <widget class="QPushButton" name="criar_regist">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>40</y>
       <width>111</width>
       <height>22</height>
      </rect>
     </property>
     <property name="text">
      <string>add register</string>
     </property>
    </widget>
    <widget class="QPushButton" name="criar_alu">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>10</y>
       <width>111</width>
       <height>22</height>
      </rect>
     </property>
     <property name="text">
      <string>add alu</string>
     </property>
    </widget>
   </widget>
   <widget class="QTreeWidget" name="opcode_display">
    <property name="geometry">
     <rect>
      <x>820</x>
      <y>0</y>
      <width>281</width>
      <height>381</height>
     </rect>
    </property>
    <property name="indentation">
     <number>10</number>
    </property>
    <property name="columnCount">
     <number>4</number>
    </property>
    <attribute name="headerDefaultSectionSize">
     <number>30</number>
    </attribute>
    <attribute name="headerMinimumSectionSize">
     <number>10</number>
    </attribute>
    <column>
     <property name="text">
      <string>op............</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>arg</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>arg</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>literal</string>
     </property>
    </column>
   </widget>
   <widget class="QTabWidget" name="repls">
    <property name="geometry">
     <rect>
      <x>510</x>
      <y>790</y>
      <width>201</width>
      <height>131</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="opcode_repl">
     <attribute name="title">
      <string>opcode</string>
     </attribute>
     <widget class="QPlainTextEdit" name="repl_display">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>191</width>
        <height>70</height>
       </rect>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QLineEdit" name="repl_input">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>80</y>
        <width>191</width>
        <height>22</height>
       </rect>
      </property>
      <property name="placeholderText">
       <string>opcode repl</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="microcode_repl">
     <attribute name="title">
      <string>microcode</string>
     </attribute>
     <widget class="QPlainTextEdit" name="microcode_repl_display">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>191</width>
        <height>70</height>
       </rect>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QLineEdit" name="microcode_repl_input">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>80</y>
        <width>191</width>
        <height>22</height>
       </rect>
      </property>
      <property name="placeholderText">
       <string>microcode repl</string>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QListWidget" name="memory_list">
    <property name="geometry">
     <rect>
      <x>820</x>
      <y>400</y>
      <width>211</width>
      <height>401</height>
     </rect>
    </property>
    <property name="modelColumn">
     <number>0</number>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton">
    <property name="geometry">
     <rect>
      <x>870</x>
      <y>380</y>
      <width>121</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>redraw memory</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_2">
    <property name="geometry">
     <rect>
      <x>810</x>
      <y>850</y>
      <width>21</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>save_state_test</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_3">
    <property name="geometry">
     <rect>
      <x>860</x>
      <y>830</y>
      <width>80</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>load_state</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="memory_fpath_input">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>880</y>
      <width>351</width>
      <height>23</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Memory path&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
    <property name="toolTipDuration">
     <number>1</number>
    </property>
    <property name="placeholderText">
     <string>Memory load path</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="opcodes_fpath_input">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>910</y>
      <width>351</width>
      <height>23</height>
     </rect>
    </property>
    <property name="placeholderText">
     <string>Opcodes load path</string>
    </property>
   </widget>
   <widget class="QPushButton" name="fetch_button">
    <property name="geometry">
     <rect>
      <x>110</x>
      <y>770</y>
      <width>80</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Fetch</string>
    </property>
   </widget>
   <widget class="QPushButton" name="execute_button">
    <property name="geometry">
     <rect>
      <x>210</x>
      <y>770</y>
      <width>80</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Execute</string>
    </property>
   </widget>
   <widget class="QPushButton" name="fetch_execute_button">
    <property name="geometry">
     <rect>
      <x>310</x>
      <y>770</y>
      <width>111</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Fetch + Execute</string>
    </property>
   </widget>
   <zorder>frame</zorder>
   <zorder>repl_cu_select</zorder>
   <zorder>repl_cu_select_label</zorder>
   <zorder>opcode_display</zorder>
   <zorder>repls</zorder>
   <zorder>memory_list</zorder>
   <zorder>pushButton</zorder>
   <zorder>pushButton_2</zorder>
   <zorder>pushButton_3</zorder>
   <zorder>memory_fpath_input</zorder>
   <zorder>opcodes_fpath_input</zorder>
   <zorder>fetch_button</zorder>
   <zorder>execute_button</zorder>
   <zorder>fetch_execute_button</zorder>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1202</width>
     <height>20</height>
    </rect>
   </property>
   <property name="defaultUp">
    <bool>true</bool>
   </property>
   <property name="nativeMenuBar">
    <bool>true</bool>
   </property>
   <widget class="QMenu" name="menuok">
    <property name="title">
     <string>ok</string>
    </property>
   </widget>
   <addaction name="menuok"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
